test/tests/bcs/dmg_periodic/tests

```diff
@@ -0,0 +1,58 @@
+[Tests]
+  issues = '#15501'
+  design = 'AddPeriodicBCAction.md  DistributedRectilinearMeshGenerator.md'
+
+  [./check]
+    type = 'RunException'
+    input = 'dmg_periodic_bc.i'
+    cli_args = '/UserObjects/uo/type=CheckGhostedBoundaries /UserObjects/uo/total_num_bdry_sides=160'
+    expect_err = 'All boundaries are ghosted to every single processor, and it is not scalable'
+    requirement = "MOOSE shall check ghosted boundaries"
+    max_parallel = 1
+  [../]
+
+  [./check_one_step]
+    type = 'Exodiff'
+    input = 'dmg_periodic_bc.i'
+    exodiff = 'dmg_periodic_bc_out_one_step.e'
+    cli_args = '/UserObjects/uo/type=CheckGhostedBoundaries /UserObjects/uo/total_num_bdry_sides=160 Outputs/hide="pid" Outputs/file_base=dmg_periodic_bc_out_one_step Executioner/num_steps=1'
+    abs_zero = 1e-6
+    rel_err = 1e-5
+    requirement = "MOOSE shall check ghosted boundaries for one time step"
+    min_parallel = 2
+  [../]
+
+  [./2d]
+    type = 'Exodiff'
+    input = 'dmg_periodic_bc.i'
+    exodiff = 'dmg_periodic_bc_out.e'
+    min_parallel = 2
+    cli_args = 'Outputs/hide="pid"'
+    abs_zero = 1e-6
+    rel_err = 1e-5
+    requirement = "MOOSE shall support periodic boundary conditions in a parallel generated mesh."
```

Comment:
This requirement is clear to me :+1:

Reply:
Maybe explicitly say that this is for 2D. I'm just guessing that's the dimension since the next two requirements are for 3D and 1D

Reply:
Yes, it is 2D